source/frontend/src/features/mapSideBar/acquisition/tabs/compensation/update/UpdateCompensationRequisitionForm.test.tsx

```diff
@@ -95,12 +118,74 @@ describe('TakesUpdateForm component', () => {
     expect(await findByText(/Detailed remarks must be at most 2000 characters/i)).toBeVisible();
   });
 
+  it('should display the payee information', async () => {
+    const compensationWithPayeeInformation = CompensationRequisitionFormModel.fromApi(
+      getMockApiCompensation(),
+    );
+    const {
+      getPayeePreTaxAmount,
+      getPayeeTaxAmount,
+      getPayeeTotalAmount,
+      getPayeeGSTNumber,
+      getPayeePaymentInTrust,
+    } = await setup({ props: { initialValues: compensationWithPayeeInformation } });
+
+    expect(getPayeePaymentInTrust()).toBeChecked();
+    expect(getPayeeGSTNumber()).toHaveValue('9999');
+    expect(getPayeePreTaxAmount()).toHaveValue('$30,000.00');
+    expect(getPayeeTaxAmount()).toHaveValue('$1,500.00');
+    expect(getPayeeTotalAmount()).toHaveValue('$31,500.00');
+  });
+
+  it('should display confirmation modal when changing the status to "FINAL"', async () => {
+    const { findByText, getByTitle, getStatusDropDown } = await setup({
+      props: { initialValues: defaultCompensation },
+    });
+
+    await act(async () => {
+      fireEvent.change(getStatusDropDown(), { target: { value: 'final' } });
+    });
+
+    expect(
+      await findByText(/You have selected to change the status from DRAFT to FINAL./i),
+    ).toBeVisible();
+
+    await act(async () => userEvent.click(getByTitle('ok-modal')));
+
+    expect(getStatusDropDown()).toHaveValue('final');
+  });
+
+  it('should return status to Draft when confirmation modal cancel', async () => {
+    const { container, findByText, getByTitle, getStatusDropDown } = await setup({
+      props: { initialValues: defaultCompensation },
+    });
+
+    await act(async () => {
+      await fillInput(container, 'status', 'final', 'select');
+    });
```

Comment:
nit: They are both equivalent but I'd prefer if you use the same method as the other tests to setting a value on the SELECT

```tsx
    await act(async () => {
      fireEvent.change(getStatusDropDown(), { target: { value: 'final' } });
    });

```

Reply:
updated